codice/ddf
# [2.13.x] DDF-4090 - Remove LDAP structure assumptions

#### What does this PR do?
Remove LDAP structure assumptions, enable LDAP login and attribute claims to work with nested user OUs, allow multiple ldap servers in a single configuration for load balancing / failover
#### Who is reviewing it? 
@ryeats @KalCramer @harrison-tarr 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams/security

#### Ask 2 committers to review/merge the PR and tag them here.

@stustison
@tbatie 
@coyotesqrl 

#### How should this be tested?
Create a user in a nested OU under the base user DN 
Set their uid and cn to be different values (e.g. uid=user1, cn=User One)
Add the user to a group and verify that they can log in and that their group information is successfully added.  If added to the admin group the user should be able to log into the admin console

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4090](https://codice.atlassian.net/browse/DDF-4090)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
